### ISSUE TITLE
What is the recommended strategy for ISVs handling cloud instances?

### ISSUE THREAD

**Issue by armstrong-charles** [microsoftgraph/microsoft-graph-docs#6946]
Thanks for the video and details. 
As per your video, following are required : Separate App ID for each instance of Cloud, End point for OAuth redirect URL, end-point for accessing Graph API 
In the above video, you showed defining different end points for each national cloud, how does it dynamically inject specific endpoints during run time from appsettings.json? If the documentation covers Microsoft recommended approach for doing this, that would be appreciated.

Just guessing the approach -- is it addressed by DevOps build process in deployment stage (or) Single binary includes different end points for each localization that refers to national cloud (global, US gov, Germany, China) endpoints based on the localization settings for Germany and China, do you handle US Government case separately? 

---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: c2dbd263-9afd-aa90-cb50-bbadfc424cc2
* Version Independent ID: 63aaf699-609e-b4dd-bd02-9f891e09db8d
* Content: [National cloud deployments - Microsoft Graph](https://docs.microsoft.com/en-us/graph/deployments)
* Content Source: [concepts/deployments.md](https://github.com/microsoftgraph/microsoft-graph-docs/blob/master/concepts/deployments.md)
* Product: **non-product-specific**
* Technology: **microsoft-graph**
* GitHub Login: @MSGraphDocsvTeam
* Microsoft Alias: **MSGraphDocsVteam**

**Comment by jthake-msft** [comment]
Duplicate of #6934 . Please ask questions on Stack Overflow and tag with [microsoft-graph] as mentioned in the guidance from the bot.

**Comment by jthake-msft** [comment]
To frame the feedback a different way.
This page should really explain how to call graph when you are not on the generic environment. We should give examples of how to use the SDKs to get the access tokens and call the graph correctly for environments like China, and US Gov for instance.

**Comment by armstrong-charles** [comment]
Thanks @jthanke-msft.
There is a note that may need some clarification. 

"Note: Apps can only access organizational data through the national cloud endpoints. This means that apps can only access data in tenants that are registered in the specific national cloud. Apps that are trying to access consumer data associated with Microsoft personal accounts through Microsoft Graph should use the global service https://graph.microsoft.com. Access tokens acquired for a national cloud deployment are not interchangeable with those acquired for the global service or any other national cloud."

The above note could imply that some apps may have access to data from the national cloud as well as the global cloud(customer data). Do you mean that? if yes, App registration for such an app where does it register with? national cloud or global cloud?
From ISVs perspective, there are 3 cases possible, cloud app deployed for National Cloud or Global Cloud, or Both (Some functionality for National Cloud that could support customers with LIVE Id).

**Comment by marcreicher** [comment]
@SonOfJesseAsSnekithan Did you ever get an answer here?

**Comment by SonOfJesseAsSnekithan** [comment]
No, although it is a kind of architecture details questions, still i believe that this docs could touch upon those scenarios.
